geopandas/geoseries.py

```diff
@@ -173,8 +222,15 @@ def copy(self, order='C'):
         copy : GeoSeries
         """
         # FIXME: this will likely be unnecessary in pandas >= 0.13
-        return GeoSeries(self.values.copy(order), index=self.index,
-                      name=self.name).__finalize__(self)
+        return GeoSeries(self._data,
+                         index=self.index, crs=self.crs,
+                         name=self.name)
```

Comment:
of course if the C objects itself are immutable, having another array (which is not a view of the original) pointing to the same C objects, might be enough to speak of a 'copy'

It's just that if you set an item in the one VectorizedGeometry, it doesn't change the other.